File: app/java/src/main/java/com/example/production/application/port/in/command/IssueExecuteCommand.java
package com.example.production.application.port.in.command;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;

/**
 * 払出実行コマンド
 */
@Data
@Builder
public class IssueExecuteCommand {
    private String instructionNumber;
    private String workOrderNumber;
    private Integer routingSequence;
    private String locationCode;
    private LocalDate issueDate;
    private String issuerCode;
    private List<IssueDetailCommand> details;
}

File: app/csharp/src/ProductionManagement.Application/Port/In/Command/IssueExecuteCommand.cs
namespace ProductionManagement.Application.Port.In.Command;

/// <summary>
/// 払出実行コマンド
/// </summary>
public class IssueExecuteCommand
{
    public required string WorkOrderNumber { get; init; }

    public required int RoutingSequence { get; init; }

    public required DateOnly IssueDate { get; init; }

    public required string IssuerCode { get; init; }

    public string? LocationCode { get; init; }

    public required IReadOnlyList<IssueDetailCommand> Details { get; init; }
}

/// <summary>
/// 払出明細コマンド
/// </summary>
public class IssueDetailCommand
{
    public required string ItemCode { get; init; }

    public required decimal IssueQuantity { get; init; }
}
